File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        // If the root is null, the tree is empty, so the depth is 0.
    if (root == null) return 0;

    // Initialize the depth level.
    int level = 0;

    // Initialize the queue and add the root node to it.
    Queue<TreeNode> queue = new LinkedList<>();
    queue.add(root);

    // Perform BFS traversal.
    while (!queue.isEmpty()) {
        // Increase the level as we are moving to a new level in the tree.
        level++;

        // Process all nodes at the current level.
        int size = queue.size();  // Get the number of nodes at this level.
        for (int i = 0; i < size; i++) {
            TreeNode current = queue.remove();

            // Add the children of the current node to the queue.
            if (current.left != null) {
                queue.add(current.left);
            }
            if (current.right != null) {
                queue.add(current.right);
            }
        }
    }

    // The final value of 'level' will be the maximum depth of the tree.
    return level;
    }
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        
        self.in_order = []

        def search(node):
            if not node: return

            search(node.left)
            self.in_order.append(node.val)
            search(node.right)

        search(root)
        return self.in_order[k - 1]